### DISCUSSION TITLE
Is there any way to disable all styles applied to root?

### DISCUSSION THREAD

**Discussion by kphamm** [saadeghi/daisyui#1452] [Q&A] [answered]
There are styles being automatically applied to my `<html>` object, is there an easy way to disable them? Specifically all the styles associated with :root

**Comment by saadeghi** [comment]
Of you tell me what exactly is the issue, I can help you better.

**Reply by kphamm** [reply]
I have a modal that I want to be able to be embeddable into any webapp, it's in React and renders at the "root" div. I just want the modal to be themed with daisyui, but it also themes the root, messing with the rest of the app that I'm importing it to.

for example, if a user is in dark mode, the website only supports light mode and the background gets changed to dark

so I just want to disable all of the root styling if possible

**Comment by saadeghi** [comment, marked as answer]
You can use this config on `tailwind.config.js`:
```js
daisyui: {
  base: false,
  themes: [],
}
```
But it disable all theme colors so you should then add the theme colors to your own wrapper like this: https://play.tailwindcss.com/p64f4sbNEu?file=css

**Reply by kphamm** [reply]
thank you! this fixed it! How can I add all the themes to the css now?

Currently I am using themes as a react state like this

` <Dialog.Panel id="my-component" data-theme={theme} />` where `theme` is one of the Daisy themes. Is there any way I can still make it compatible with every daisy theme?

**Reply by saadeghi** [reply]
Yes, you can manually copy all the themes (or the ones you want) [from here](https://unpkg.com/browse/daisyui@2.46.0/dist/themes.css) and put them in your CSS.

All themes except the default one at the start of file is based on the `data-theme` attribute.  
The first one is for `:root` which would be the default theme (if there are no `data-theme` attribute)  
So if your component has a mandatory `data-theme` you can safely remove that `:root` one.

**Reply by kphamm** [reply]
Thank you so much!
